### ISSUE TITLE
Possibly incorrect denom name validation

### ISSUE THREAD

**Issue by jstuczyn** [cosmos/cosmos-rust#173]
Hi again!

While browsing through the code I encountered this comment in the `FromStr` implementation for `Denom`: https://github.com/cosmos/cosmos-rust/blob/ed05f6909cfdf97609e2f3d8b618fd0244a2d13d/cosmrs/src/base.rs#L178-L189

I think the validation might actually be incorrect. My argument for that is that I sent a `QueryTotalSupplyRequest` request, thus receiving the corresponding `QueryTotalSupplyResponse`. To remind, it looks as following:
```rust
/// QueryTotalSupplyResponse is the response type for the Query/TotalSupply RPC
/// method
#[derive(Clone, PartialEq, ::prost::Message)]
pub struct QueryTotalSupplyResponse {
    /// supply is the supply of the coins
    #[prost(message, repeated, tag = "1")]
    pub supply: ::prost::alloc::vec::Vec<super::super::base::v1beta1::Coin>,
    /// pagination defines the pagination in the response.
    #[prost(message, optional, tag = "2")]
    pub pagination: ::core::option::Option<super::super::base::query::v1beta1::PageResponse>,
}
```

As you can see, the response contains proto representation of a cosmos `Coin`. Logically I attempted to parse it into `cosmrs::Coin` type, which failed as the denom was incorrect. It makes perfect sense given the existing validation. Now, as for the value returned from the query:
```rust
QueryTotalSupplyResponse {
    supply: [
        Coin {
            denom: "ibc/9F53D255F5320A4BE124FF20C29D46406E126CE8A09B00CA8D3CFF7905119728",
            amount: "1500",
        },
        Coin {
            denom: "unymt",
            amount: "1000000000000000",
        },
        Coin {
            denom: "unyxt",
            amount: "999999996000000",
        },
    ],
    pagination: None,
}
```

The denom established via the IBC channel, i.e. `ibc/9F53D255F5320A4BE124FF20C29D46406E126CE8A09B00CA8D3CFF7905119728` failed to get parsed. Now, the "possibly" part of the title of the issue comes from the fact that I'm unsure whether this is expected Cosmos' behaviour or not, i.e. whether IBC denoms are "valid" in this context. But I thought I might let you know of this : )

**Comment by tony-iqlusion** [comment]
Yes that's definitely a valid concern! I'll loosen the constraints

**Comment by tony-iqlusion** [comment]
Fix for this: #182
